### PR TITLE
Fix potential NullPointerException in UserContext by adding null check for parentContext.

### PR DESCRIPTION
**Pull request by mrkartik00**
Title: [#4331] fix: add null check for parentContext in UserContext

### What changes were proposed in this pull request?

This pull request adds a null check for the parentContext variable in the getUserContext method of UserContext.java. This ensures that a NullPointerException is avoided when deepCopy() is called on parentContext.

### Why are the changes needed?

The change is needed to prevent a potential NullPointerException if parentContext is null. Adding this check makes the code more reliable.

Fix: #4331 

### Does this PR introduce _any_ user-facing change?

No, this change does not introduce any user-facing changes.

### How was this patch tested?

The change was reviewed to ensure that the null check effectively prevents the issue. No new tests were needed, as this is a preventive fix.

### CODE REVIEW

**Review by justinmclean — COMMENTED** [review]
I would leave the existing comment as is and not remove it

File: catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/UserContext.java [hunk 1]
```diff
@@ -123,9 +123,8 @@ public static UserContext getUserContext(
       addUserContext(nameIdentifier, simpleUserContext);
       return simpleUserContext;
     } else if (authenticationType == AuthenticationType.KERBEROS) {
-      // if the kerberos authentication is inherited from the parent context, we will use the
-      // parent context's kerberos configuration.
```
Comment (yuqi1129) [inline catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/UserContext.java:127]: You'd better keep those comments.

**Comment by justinmclean** [comment]
@mrkartik00 It looks like the comments have still been removed. Can you please add them back?

**Comment by justinmclean** [comment]
Please remove the "// Improvement: Check if parentContext is not null before using it" there is no need for this comment in the code.

File: catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/UserContext.java [hunk 2]
```diff
@@ -125,7 +125,7 @@ public static UserContext getUserContext(
     } else if (authenticationType == AuthenticationType.KERBEROS) {
       // if the kerberos authentication is inherited from the parent context, we will use the
       // parent context's kerberos configuration.
-      if (authenticationConfig.isSimpleAuth()) {
+      if (parentContext != null && authenticationConfig.isSimpleAuth()) {
```
Comment (yuqi1129) [inline catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/UserContext.java:128]: In fact, if the value of `authenticationType` is `KERBEROS` and `authenticationConfig.isSimpleAuth()` is true, the value of `parentContext` is not null.  Anyway, adding a check won't make things worse.

**Review by yuqi1129 — APPROVED** [review]

**Pull request merged**